torch/csrc/distributed/rpc/utils.cpp

```diff
@@ -538,26 +528,30 @@ std::tuple<tensorpipe::Message, TensorpipeWriteBuffers> tensorpipeSerialize(
   pickler.protocol();
   pickler.pushIValue(buffers.tensors);
   pickler.stop();
-  // kTpMessagePickleIdx = 4
+  // kTpMessagePickleIdx = 3
   tpMessage.payloads.push_back(tensorpipe::Message::Payload{
       buffers.pickle.data(), buffers.pickle.size()});
-  for (const auto& tensor : pickler.tensorData()) {
-    const auto& tensorData = jit::getWriteableTensorData(tensor);
+  for (size_t i = 0; i < pickler.tensorData().size(); ++i) {
+    const auto& tensorData =
+        jit::getWriteableTensorData(pickler.tensorData()[i]);
```

Comment:
Is `pickler.tensorData()` an efficient function that we can call repeatedly at no cost? Or should we store its value (or a reference to it)?

Reply:
Yep, it is an efficient function. Let me create a reference anyway incase it changes in the future.

https://github.com/pytorch/pytorch/blob/af9cad761ad1dd215470ebce58932b82d6ba8bee/torch/csrc/jit/serialization/pickler.h#L145-L147